### PR TITLE
doc: ByteBuffer improvements

### PR DESCRIPTION
**Pull request by ianpartridge**
A few improvements to the docs for `ByteBuffer`.

### Motivation:

Improve the docs.

### Modifications:

- `ByteBufferAllocator().buffer` is not static.
- Datatypes in get<type> should be capitalised like `Data` already is.
- `readerIndex`/`writerIndex` should be documented under sequential access, not random access.
- `changeCapacity()` parameter is `to` not `capacity`.

### Result:

More accurate documentation.

### CODE REVIEW

File: Sources/NIO/ByteBuffer-core.swift [hunk 1]
```diff
@@ -87,7 +87,7 @@ private func toIndex(_ value: Int) -> Index {
 /// more bytes (octets).
 ///
 /// ### Allocation
-/// Use `ByteBufferAllocator.buffer(capacity: desiredCapacity)` to allocate a new `ByteBuffer`.
+/// Use `ByteBufferAllocator().buffer(capacity: desiredCapacity)` to allocate a new `ByteBuffer`.
```
Comment (normanmaurer) [inline Sources/NIO/ByteBuffer-core.swift]: @ianpartridge maybe we should make it clear that you usually should not create a new `ByteBufferAllocator` for each allocation ? What about we make it `allocator.buffer(...)` ?

**Comment by normanmaurer** [comment]
@swift-nio-bot test this please

**Comment by normanmaurer** [comment]
@ianpartridge let me know once addressed and rebased .

**Comment by ianpartridge** [comment]
@normanmaurer Done.

**Comment by Lukasa** [comment]
@swift-nio-bot ok to test

**Pull request merged**